source/extensions/filters/common/lua/wrappers.h

```diff
@@ -112,10 +112,145 @@ class MetadataMapWrapper : public BaseLuaObject<MetadataMapWrapper> {
  */
 class SslConnectionWrapper : public BaseLuaObject<SslConnectionWrapper> {
 public:
-  SslConnectionWrapper(const Ssl::ConnectionInfoConstSharedPtr) {}
-  static ExportedFunctions exportedFunctions() { return {}; }
+  SslConnectionWrapper(const Ssl::ConnectionInfoConstSharedPtr info) : connection_info_{info} {}
```

Comment:
`explicit`

Should we be passing a reference here instead of a ptr to avoid having to assert on it being non null? Or are there lifetime concerns?

Reply:
Hum, since `info` can be `nullptr` here and semantically we want to be able to check that if `streamInfo.dowstreamSslConnection()` is `nil` from a Lua script, that means the connection is not secured.

Reply:
> Should we be passing a reference here instead of a ptr to avoid having to assert on it being non-null? Or are there lifetime concerns?

Thanks for pointing this out. I made it as a ref now.